service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterMapper.java

```diff
@@ -48,48 +49,58 @@ public String mapEncounterToEhrComposition(Encounter encounter) {
-            .code(buildCode(encounter));
+            .code(buildCode(encounter))
+            .displayName(buildDisplayName(encounter))
+            .originalText(buildOriginalText(encounter));
 
         return TemplateUtils.fillTemplate(ENCOUNTER_STATEMENT_TO_EHR_COMPOSITION_TEMPLATE,
             encounterStatementTemplateParameters.build());
     }
 
-    private String buildCode(Encounter encounter) {
+    private boolean isSnomedAndWithinEhrCompositionVocabularyCodes(Coding coding) {
+        return coding.hasSystem() && coding.getSystem().equals(SNOMED_SYSTEM)
+            && EHR_COMPOSITION_NAME_VOCABULARY_CODES.contains(coding.getCode());
+    }
+
+    private Optional<Coding> extractCoding(Encounter encounter) {
         var type = encounter.getType()
             .stream()
             .findFirst()
             .orElseThrow(() -> new EhrMapperException("Could not map Encounter type"));
 
-        var coding = type.getCoding()
+        return type.getCoding()
             .stream()
             .filter(this::isSnomedAndWithinEhrCompositionVocabularyCodes)
             .findFirst();
+    }
 
+    private String buildCode(Encounter encounter) {
+        var coding = extractCoding(encounter);
         if (coding.isPresent()) {
-            return String.format(EHR_COMPOSITION_SNOMED_CODE, coding.get().getCode(), coding.get().getDisplay(), buildCodeText(type));
-        } else {
-            return String.format(EHR_COMPOSITION_OTHER_REPORT_CODE, buildCodeText(type));
+            return coding.get().getCode();
         }
+        return OTHER_REPORT_CODE;
     }
 
-    private boolean isSnomedAndWithinEhrCompositionVocabularyCodes(Coding coding) {
-        return coding.hasSystem() && coding.getSystem().equals(SNOMED_SYSTEM)
-            && Arrays.asList(EHR_COMPOSITION_NAME_VOCABULARY_CODES).contains(coding.getCode());
+    private String buildDisplayName(Encounter encounter) {
+        var coding = extractCoding(encounter);
+        if (coding.isPresent()) {
+            return coding.get().getDisplay();
+        }
+        return OTHER_REPORT_DISPLAY;
     }
 
-    private String buildCodeText(CodeableConcept codeableConcept) {
-        if (codeableConcept.hasText()) {
-            return String.format(EHR_COMPOSITION_CODE_TEXT, codeableConcept.getText());
-        } else {
-            return StringUtils.EMPTY;
+    private String buildOriginalText(Encounter encounter) {
+        if (encounter.getTypeFirstRep().hasText()) {
+            return encounter.getTypeFirstRep().getText();
         }
+        return StringUtils.EMPTY;
     }
 
     @SneakyThrows
-    private static String[] getEhrCompositionNameVocabularyCodes() {
-        InputStream inputStream = EncounterMapper.class.getClassLoader().getResourceAsStream("ehr_composition_name_vocabulary_codes.txt");
-        String ehrCompositionNameCodes = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+    private static HashSet<String> getEhrCompositionNameVocabularyCodes() {
```

Comment:
A few things to consider here:
* Use the interface (Map) instead of the implementation (HashMap) wherever possible
* Creating the InputStream as a static variable is a bad idea. Nearly all streams need to be closed when you're finished when them. In modern version of java these are "auto-closable" when used in a try clause. In the past you needed to try{...} catch{ ...} finally { stream.close(); }
* Avoid doing things yourself (e.g. splitting by a line) when the standard libraries / language will do this for you. The code might not work as written if the file happens to use windows style line endings.
* Avoid creating objects you don't need. Here we create an array, then a list, and then a hashset. How about just creating the hashmap and adding the elements to it. The Streams API gives us a nice way to collect items into a set.
* 
```
@SneakyThrows
    private static Set<String> getEhrCompositionNameVocabularyCodes() {
        try (InputStream is = EncounterMapper.class.getResourceAsStream("ehr_composition_name_vocabulary_codes.txt")) {
            BufferedReader reader = new BufferedReader(new InputStreamReader(is));
            return reader.lines().collect(Collectors.toSet());
        }
    }
````

Reply:
This could be improved further by filtering blank lines:

```
return reader.lines()
                .filter(String::isBlank)
                .collect(Collectors.toSet());
```

Otherwise there is a risk that a blank or whitespace code would be treated as a valid input.

Reply:
Finally, upon inspecting the API documentation for the Collectors::toSet method, there is no promise about what kind of set it returns. We do however have an option to create an unmodifiableSet:

```
return reader.lines()
                .filter(String::isBlank)
                .collect(Collectors.toUnmodifiableSet());
```

Reply:
Have implemented these changes. 
Needed to add a suppression on RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE to use try-with-resources as there is a known issue with Spotbugs giving false positives.
https://github.com/spotbugs/spotbugs/issues/1338